### DISCUSSION TITLE
Issue with config: matchUpdateTypes breaks it?

### DISCUSSION THREAD

**Discussion by frykten** [renovatebot/renovate#29274] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github, renovatebot/github-action@v40.1.1

### Please tell us more about your question or problem

We recently updated our renovate config with some huge overhaul: we removed PRs for majors, added automerge, refined groups, _etcetera_.

However, this new config, as simple as it may be, did not work and stopped creating PRs. We then removed all configurations and tried debugging the whole step by step. In the end, even the first step would not work: on a simple configuration, using the matchUpdateTypes property blocked it (without, PRs were created, with they would keep in the void, the dashboard would remain empty).

We had that configuration:

```
{
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "branchPrefix": "renovate/",
   "baseBranches": [
      "sprint"
   ],
   "dependencyDashboard": true,
   "repositories": [
      "redacted"
   ],
   "lockFileMaintenance": {
      "enabled": false
   },
   "packageRules": [
      {
         "matchPackagePatterns": [
            "*"
         ],
         "enabled": false
      },
      {
         "matchPackagePatterns": [
            "@radix-ui"
         ],
         "groupName": "Radix packages",
         "matchUpdateTypes": [
            "digest",
            "patch",
            "minor"
         ],
         "enabled": true
      }
   ],
   "prConcurrentLimit": 5,
   "rebaseWhen": "behind-base-branch",
   "reviewers": [
      "team:frontend"
   ],
   "vulnerabilityAlerts": {
      "enabled": true
   }
}
```

Removing the matchUpdateTypes property updated the dashboard, and created the PRs.

Supposedly, this should work?

**Comment by rarkins** [comment]
Your earlier rule disables all packages before they are looked up.  If they're not looked up then you don't get any updates to match against.

**Reply by frykten** [reply]
We had thought (and so far it has worked as such) that the first rules were overwritten by the latest ones? Such as:

- disable all
- least important rules
- mildly important ones
- most important ones

Following the documentation and tutorials the "match all and disable all" first rule always come first. And it has worked quite well so far? Why would it break only the "matchUpdateTypes"?

**Reply by RahulGautamSingh** [reply]
Renovate flow: extract -> lookup -> update
We apply package rules at the end of extract phase and multiple times during the lookup phase.

Now, `updateType` is only initialized in the lookup phase ie. the second pkg rule you have will not be matched during extract phase.
Only, the first pkg rule will be matched, which will disable all the deps.

If the deps are disabled we will not lookup them.

Your other configs might have worked because in the second pkg rule you must have some other field to match against eg. `matchPackageNames` which would have then enabled those packages. Hence, they were looked up and pr created.

**Reply by rarkins** [reply, marked as answer]
Here's how Renovate works:
- It extracts existing dependencies
- It looks up new versions
- It applies updates

Package rules are applied essentially before/after each step.

Your first rule disables all dependencies after the first step. There is then no second step if all dependencies are disabled. Your second rule could only be applied if there were any updates from the lookup stage. It never gets applied.

If Renovate were to do what you expect it to, it would mean that it would need to go and look up every disabled dependency just in case there's a matching rule "later". Which means users cannot disable lookups, which would be bad.

Instead you should rewrite your rules like this:
- Modify your first rule to exclude @radix-ui from the disabling
- Remove enabled=true from the second rule as it's no longer needed
- Add a third rule which disables major updates

**Reply by frykten** [reply]
Thank you for your answer, both of you.

We are not sure we understand what you mean: currently, we have many many rules, which all work (we have a dashboard fully setup, with only the packages we wish to update, and the others are well disabled). The ONLY issue we have is when we use the matchUpdateTypes property; when we add this property, then nothing works anymore.

Is that property changing something in the flow? Should we share more of our config so you can tell us what is wrong?

**Reply by rarkins** [reply]
Unfortunately I think we've reached the reasonable limits of how much we can do in this forum. I have gone to long lengths to explain things the best I can in my last comment, and I have no more to add.

Regarding our project policy for matters like this please read https://github.com/renovatebot/renovate/blob/main/CODE_OF_CONDUCT.md#if-you-have-urgent-work and the heading below it "Getting more help".
